### PR TITLE
[feat] Products filters

### PR DESCRIPTION
**Pull request by shivampandey0**
## Filters for Product Page
- Range Slider component 
- FilterInput component for Radio and Checkbox
- Sort by Price & Rating
- Filter by Product Types
- Filter by Categories
- Rating Filter
- Filter stocked out products
- Filter Fast Delivery eligible products
- Clear all Filters
- Product Types endpoint in backend

https://user-images.githubusercontent.com/51923995/159177805-4f61a4e1-f984-489d-b8a8-2060c9de07a3.mp4

### CODE REVIEW

File: src/backend/controllers/TypeController.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Response } from "miragejs";
+
+/**
```
Comment (hinanshisuthar) [inline src/backend/controllers/TypeController.js:3]: Would suggest to remove comments that came with pre-installation

File: src/backend/db/types.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { v4 as uuid } from "uuid";
+
+/**
```
Comment (hinanshisuthar) [inline src/backend/db/types.js:3]: Here too the comments can be removed

**Review by hinanshisuthar — APPROVED** [review]
Code structure looks good. Broken everything into smaller components was nice.

File: src/backend/controllers/TypeController.js (hunk 1 above)
Comment (shivampandey0) [reply src/backend/controllers/TypeController.js:3]: These comments came with backend and help as doc.

File: src/backend/db/types.js (hunk 2 above)
Comment (shivampandey0) [reply src/backend/db/types.js:3]: These comments came with backend and help as doc.

File: src/pages/products/Filters.jsx [hunk 3]
```diff
@@ -1,93 +1,250 @@
...
+                  }
+                />
+              );
+            })}
+          </fieldset>
+
+          <fieldset className="my-2">
+            <legend className="fw-bold">Rating</legend>
+
+            <FilterInput
```
Comment (GuptaEesh) [inline src/pages/products/Filters.jsx]: You can take an array [1,2,3,4,5] and write the ratings logic inside it.
This will help reduce the code.

**Review by GuptaEesh — COMMENTED** [review]
Code looks great.
Good naming conventions used.
Great folder structure.
Good use of barrel imports.
Just added a small suggestion.

File: src/pages/products/Filters.jsx (hunk 3 above)
Comment (shivampandey0) [reply src/pages/products/Filters.jsx]: Done

**Pull request merged**